### ISSUE TITLE
bug: programs.git-credential-oauth eval error: cannot coerce a set to a string

### ISSUE THREAD

**Issue by powpingdone** [nix-community/home-manager#6005]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

When `programs.git-credential-oauth.enable = true;`,  `nixos-rebuild` fails with the following error:
```nix
         at /nix/store/a8yg0d34q9qfkc693p0mc2g12v5n50vp-source/modules/programs/git-credential-oauth.nix:33:52:
           32|     programs.git.extraConfig.credential.helper = [
           33|       ("${cfg.package}/bin/git-credential-oauth" + lib.mkIf cfg.extraFlags
             |                                                    ^
           34|         " ${lib.strings.concatStringsSep " " cfg.extraFlags}")

       error: cannot coerce a set to a string: { _type = "if"; condition = «thunk»; content = «thunk»; }

```

### Maintainer CC

@tomodachi94 

### System information

```markdown
- system: `"aarch64-linux"`
 - host os: `Linux 6.12.0-rc4, NixOS, 24.11 (Vicuna), 24.11.20241018.4c2fcb0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/riqkpszjqk02bi1wppfg8ip5xvh102qd-source`
```

**Comment by tomodachi94** [comment]
😅 This is what I get for not writing tests. Fix (and a test case) coming soon.

**Comment by tomodachi94** [comment]
Fix created at #6008. I also added some tests to avoid something as obvious as this from slipping through again 🙈

**Comment by tomodachi94** [comment]
See #6008
